crates/runtime/README.md

```diff
@@ -0,0 +1,662 @@
+
+  - [Introduction](#introduction)
+  - [Core components](#core-components)
+    - [Overview](#overview)
+    - [VMLogic](#vmlogic)
+    - [Host functions](#host-functions)
+    - [Resource limits](#resource-limits)
+    - [Storage](#storage)
+    - [Error handling](#error-handling)
+    - [Serialisation and deserialisation](#serialisation-and-deserialisation)
+  - [Operational flow](#operational-flow)
+  - [Detailed component descriptions](#detailed-component-descriptions)
+    - [Wasm module execution (`run()` function)](#wasm-module-execution-run-function)
+    - [VMLogic](#vmlogic-1)
+    - [Host functions](#host-functions-1)
+    - [Memory management](#memory-management)
+    - [Storage](#storage-1)
+    - [Error handling](#error-handling-1)
+  - [Usage examples](#usage-examples)
+
+## Introduction
+
+The Calimero runtime is a lightweight, sandboxed execution environment for
+WebAssembly (Wasm) modules. It provides a secure and controlled platform for
+running untrusted code with defined resource limits and a set of host-provided
+functions. The runtime is designed to facilitate the execution of smart
+contracts or similar applications in a blockchain-like environment. Its modular
+design allows for easy extension and customisation, making it suitable for
+various applications requiring sandboxed execution of untrusted code.
+
+Key features of the Calimero runtime include:
+
+  - Secure execution of Wasm modules
+  - Resource limitation and metering
+  - Host-provided functions for storage, logging, and external interactions
+  - Support for deterministic execution
+  - Error handling and panic recovery
+
+This document provides a [high-level summary](#core-components) of the runtime's
+architecture and core components, followed by a [detailed description of each
+component](#detailed-component-descriptions) . It also includes a brief overview
+of the [operational flow](#operational-flow), and [usage examples](#usage-examples).
+
+## Core components
+
+### Overview
+
+```mermaid
+graph TD
+    A[Wasm Module] --> B[Calimero Runtime]
+    B --> C[VMLogic]
+    C --> D[Host Functions]
+    C --> E[Storage]
+    C --> F[Resource Limits]
+    C --> G[Error Handling]
+    D --> H[Memory Operations]
+    D --> I[Register Operations]
+    D --> J[Storage Operations]
+    D --> K[Logging/Events]
+    D --> L[External Requests]
+    E --> M[InMemoryStorage]
+    F --> N[VMLimits]
+```
+
+### VMLogic
+
+The `VMLogic` struct is the central component of the runtime. It encapsulates:
+
+  - Storage access
+  - Memory management
+  - Context information
+  - Resource limits
+  - Registers for inter-function communication
+  - Logging and event emission
+
+*[Read more...](#vmlogic-1)*
+
+### Host functions
+
+The runtime provides a set of host functions that can be called from the guest
+Wasm module:
+
+  - Memory and register operations
+  - Storage read/write
+  - Logging and event emission
+  - Cryptographic operations (keypair creation)
```

Comment:
Ah, my bad! Is this something we will be adding in...? I would have thought it would be under our remit - do we expect people to always have and provide a keypair?

Reply:
Keypair creation and it's management happen outside the runtime, the runtime will eventually expose signing operations, but it's missing right now. You can retain it if you mention signing instead.